worker-message-prioritization-distribution/src/main/java/com/github/workerframework/workermessageprioritization/redistribution/lowlevel/StagingQueueConsumer.java

```diff
@@ -38,6 +41,19 @@ public StagingQueueConsumer(final Channel channel, final StagingQueueTargetQueue
         this.stagingQueueTargetQueuePair = stagingQueueTargetQueuePair;
     }
 
+    public void recordCancellationRequested() {
+        this.cancellationRequestSentTime = Optional.of(Instant.now());
```

Comment:
If cancellation was previously requested should we allow the cancellationRequestSentTime to be reset?